src/pages/VenuePage/TemplateWrapper.tsx

```diff
@@ -15,13 +15,16 @@ import { WithNavigationBar } from "components/organisms/WithNavigationBar";
 import { PartyMap } from "components/templates/PartyMap";
 import { Jazzbar } from "components/templates/Jazzbar";
 import { AnnouncementMessage } from "components/molecules/AnnouncementMessage";
+import UserProfileModal from "components/organisms/UserProfileModal";
+import { useProfileModal } from "hooks/useProfileModal";
 
 type TemplateWrapperProps = {
   venue: Venue;
 };
 
 const TemplateWrapper: React.FC<TemplateWrapperProps> = ({ venue }) => {
   const history = useHistory();
+  const { selectedUserProfile, setUserProfile } = useProfileModal();
```

Comment:
You can also have `unsetUserProfile` inside this hook. This way you will not need to expose the logic of how you do it below.

Reply:
And it should be wrapped in `useCallback` so that it has a stable function reference when passed as props to `onHide`/etc, to prevent unnecessary re-renders.